ci(translation): save screenshots as workflow artifacts

cfm opened the pull request:
## Status

Ready for review

## Description of Changes

Fixes #7237 by saving the screenshots generated by `make translation-test` as CI artifacts for 24 hours, for the following reasons:
1. `make translation-test` generates them no matter what, so we might as well make them inspectable.
2. #7237 occurs only locally, where this code path is rarely used; let's take advantage of CI exercising it.
3. This points towards perhaps even updating Weblate's copy of screenshots directly from these tests run nightly on `develop`.

Fixes #7241 by correcting the parameterization of tests and fixtures for page-layout tests.  See individual commits for details.

## Testing

- [x] Visual review.
- [x] Inspect a couple of languages and confirm that their ~~screenshots~~ Journalist Interface screenshots are correct.
    - The Source Interface is deferred to #7354.
- ~24 hours is a reasonable retention period for 100 MB of screenshots per run.~

## Deployment

CI-only; no deployment considerations.

cfm committed:
chore: un-special-case en_US as a supported language

Only languages listed in "i18n.json" are reported by "make
supported-languages", included in the CI matrix, and usable as $LANG in
"LOCALES=$LANG make test".

cfm committed:
ci: save translation screenshots as workflow artifacts

This is about 4 MB × n languages, currently < 100 MB per run.

legoktm commented on .github/workflows/translation.yml:
```diff
@@ -56,3 +56,9 @@ jobs:
+        with:
+          name: screenshots-${{ matrix.locale }}
+          path: securedrop/tests/functional/pageslayout/screenshots/
+          retention-days: 1
```
Given that this job is run pretty rarely, I think 30 or even the default of 90 days is probably better. I'd only want to limit it if we start running out of space.

1 is too short given that no one reviewed this PR in time and they've already expired 🙃

cfm replied:
I've gotten rid of `retention-days` altogether in b3807124b86d8bc3dde25aeb69dad10df895c0b7.

legoktm reviewed (changes requested):
See inline comment.

I retriggered CI so there are artifacts that can be downloaded from https://github.com/freedomofpress/securedrop/actions/runs/11190459190 - but when I looked at some of the `el` and `de_DE` screenshots, they were all in English. (Or did I just happen to look at the wrong screenshots?)

cfm commented:
Good catch, @legoktm.  I'm looking into this in #7241.

cfm committed:
test(got_expected_language): html[lang] should be one of the languages sent in the "Accept-Languages" header

cfm committed:
fix: configure SUPPORTED_LOCALES=$TEST_LOCALES in all fixtures

Otherwise they provide only the default SUPPORTED_LOCALES =
[FALLBACK_LOCALE] = ["en_US"].

cfm committed:
fix(firefox_web_driver): configure Firefox's locale at fixture time

Otherwise the fixture isn't set up to request the language expected by
the tests that use it.

cfm commented:
The fixes are in as of <https://github.com/freedomofpress/securedrop/issues/7241#issuecomment-2398211481>.  Two cheers for putting rarely-exercised code paths in CI!  Back to you, @legoktm.

cfm committed:
test: don't search for source strings when testing a translation

And run all Journalist Interface page-layout tests in Firefox.

cfm committed:
refactor: move locale parameterization from tests to the firefox_web_driver fixture

That is, instead of receiving a "locale" fixture directly, tests that
use the "firefox_web_driver" fixture are now parameterized indirectly,
via the firefox_web_driver.locale attribute received from the
parameterized fixture.

cfm commented:
I wrestled with this some more today and found another problem with how the `firefox_web_driver` fixture was parameterized, which I've refactored in 7900e21b6326e6b86cb0ee145a15bed7f9fb67ed.  Back to you for real, @legoktm.

legoktm commented:
When I download `screenshots-de_DE` from https://github.com/freedomofpress/securedrop/actions/runs/11300734349, I see three subfolders, `de-DE`, `de_DE`, and `en-US`. So somewhere the dash is being normalized differently?

And...the de screenshots are still in English :/

cfm commented:
The good news is that the screenshots are being captured correctly.  The bad news is that the tests themselves still aren't (and haven't been) parameterized fully with the intended locale.  I'm looking into it.

cfm committed:
fix(_create_firefox_driver): set "intl.accept_languages" via FirefoxOptions rather than FirefoxProfile

<https://stackoverflow.com/a/71766991> claims that FirefoxProfile is in
fact deprecated as of Selenium 4.  I can't find any obvious evidence of
that, but it works.

cfm commented:
I found a couple more fundamental parameterization problems yesterday and today, for which I've just force-pushed fixes.  I'm going to monitor `translation-tests` for failing string checks before this is ready for another review.

legoktm commented:
I'd be happy to approve parts of this in separate PRs if that would be easier for you. e.g. the saving as artifacts part is totally fine and good to go, same with some of the other intermediate fixes.

cfm commented:
Thanks, @legoktm.  As of 6c9917c771187bf9ac4c08dc46df35d367245986 I've now got this sorted for the Journalist Interface in Firefox.  Tomorrow I should be able to fix up the Source Interface in Tor Browser the same way.  If not, I'll split it between the two interfaces.

cfm commented:
Ready here for the Journalist Interface.  The Source Interface I've deferred to #7354.

legoktm reviewed (approved):
Awesome, JI side looks good; thanks!
